### PR TITLE
nix: revive nix_2_6

### PR DESCRIPTION
**Pull request by Profpatsch**
The critical regression bug in
https://github.com/NixOS/nix/issues/6572 does not appear (at least as often) in nix 2.6 according to the discussion there, so we need to allow people to pin this older version.

Partial revert of
https://github.com/NixOS/nixpkgs/commit/b2eea05b06baa2886039d61485d37c338ad3d578

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -50,7 +50,10 @@ in lib.makeExtensible (self: {
 
   nix_2_5 = throw "nixVersions.nix_2_5 has been removed";
 
-  nix_2_6 = throw "nixVersions.nix_2_6 has been removed";
+  nix_2_6 = common {
```
Comment (thiagokokada) [inline pkgs/tools/package-management/nix/default.nix:53]: Maybe we should add a comment pointing for the bug so we don't remove this until we have it fixed.

Comment (SuperSandro2000) [reply pkgs/tools/package-management/nix/default.nix:53]: Yes, please otherwise the next person it going to drop it again.

Comment (Profpatsch) [reply pkgs/tools/package-management/nix/default.nix:56]: I added a comment

**Review by thiagokokada — APPROVED** [review]

**Comment by roberth** [comment]
Will be fixed properly soon with https://github.com/NixOS/nix/pull/7283

**Comment by Artturin** [comment]
2_6 will be removed again in https://github.com/NixOS/nixpkgs/pull/204481

**Pull request merged**